src/vm/readytoruninfo.cpp

```diff
@@ -53,7 +53,7 @@ MethodDesc * ReadyToRunInfo::GetMethodDescForEntryPoint(PCODE entryPoint)
     }
     CONTRACTL_END;
 
-#ifdef _TARGET_AMD64_
+#if defined(_TARGET_AMD64_) || defined(_TARGET_X86_)
```

Comment:
This should be enabled only for _TARGET_X86_ only for Unix (FEATURE_PAL). Windows x86 doesn't have this problem.